cmd/build.go

```diff
@@ -92,6 +104,13 @@ func ValidNamespaceAndRegistry(path string) survey.Validator {
 	}
 }
 
+// KnownBuilders are a typed string slice of builder short names which this
+// CLI understands.  Includes a customized String() representation intended
+// for use in flags and help text.
+func KnownBuilders() builders.Known {
+	return builders.Known([]string{builders.Pack, builders.S2I})
```

Comment:
Shouldn't this use `builders.All()`?
```suggestion
	return builders.Known([]string{builders.Pack, builders.S2I})
```

Reply:
Yes, we could, but that isn't _exactly_ reflecting the intention and may lead to confusion later.  The point is a bit nuanced but worth perhaps taking a swing at: I was trying to emphasize that this function is where the exact set of builders are reported _which the CLI supports_ (can create concrete implementations for when used).  The `builders.All` returns the set of _known_ builder short names globally for map references, etc.  These are actually not the same set.

While it's true that in our func CLI, those two sets are currently equivalent, I feel like writing it in this way emphasizes the point that that they are not:  a user of the Client package may have less, or more, that that "known builders".  This is also why the builder validator is not in `builders`.  By skipping straight to `return builders.Known(builders.All())`, I felt that ran the risk of leading a reader to incorrectly presume that the set of known builder _names_ and the set of builders supported by a user (the cli in this case) of the client library are necessarily equivalent.

This may be too pedantic, and does look like an oversight.  Still think I should change after reading the justification?